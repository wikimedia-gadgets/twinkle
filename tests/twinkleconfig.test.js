'use strict';

describe('modules/twinkleconfig', () => {
	describe('Twinkle.config global variables', () => {
		test('Should match snapshot', () => {
			expect(Twinkle.config.commonSets.csdCriteria).toMatchSnapshot();
			expect(Twinkle.config.commonSets.csdCriteriaDisplayOrder).toMatchSnapshot();
			expect(Twinkle.config.commonSets.csdCriteriaNotification).toMatchSnapshot();
			expect(Twinkle.config.commonSets.csdCriteriaNotificationDisplayOrder).toMatchSnapshot();
			expect(Twinkle.config.commonSets.csdAndImageDeletionCriteria).toMatchSnapshot();
			expect(Twinkle.config.commonSets.csdAndImageDeletionCriteriaDisplayOrder).toMatchSnapshot();
		});
	});
});
