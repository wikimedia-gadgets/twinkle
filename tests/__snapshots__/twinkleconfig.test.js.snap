// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/twinkleconfig Twinkle.config global variables Should match snapshot 1`] = `
{
  "a1": "A1",
  "a10": "A10",
  "a11": "A11",
  "a2": "A2",
  "a3": "A3",
  "a7": "A7",
  "a9": "A9",
  "c1": "C1",
  "c4": "C4",
  "db": "Custom rationale ({{db}})",
  "f1": "F1",
  "f2": "F2",
  "f3": "F3",
  "f7": "F7",
  "f8": "F8",
  "f9": "F9",
  "g1": "G1",
  "g10": "G10",
  "g11": "G11",
  "g12": "G12",
  "g13": "G13",
  "g14": "G14",
  "g2": "G2",
  "g3": "G3",
  "g4": "G4",
  "g5": "G5",
  "g6": "G6",
  "g7": "G7",
  "g8": "G8",
  "r2": "R2",
  "r3": "R3",
  "r4": "R4",
  "t5": "T5",
  "u1": "U1",
  "u2": "U2",
  "u5": "U5",
  "x3": "X3",
}
`;

exports[`modules/twinkleconfig Twinkle.config global variables Should match snapshot 2`] = `
[
  "db",
  "a1",
  "a2",
  "a3",
  "a7",
  "a9",
  "a10",
  "a11",
  "c1",
  "c4",
  "f1",
  "f2",
  "f3",
  "f7",
  "f8",
  "f9",
  "g1",
  "g2",
  "g3",
  "g4",
  "g5",
  "g6",
  "g7",
  "g8",
  "g10",
  "g11",
  "g12",
  "g13",
  "g14",
  "r2",
  "r3",
  "r4",
  "t5",
  "u1",
  "u2",
  "u5",
  "x3",
]
`;

exports[`modules/twinkleconfig Twinkle.config global variables Should match snapshot 3`] = `
{
  "a1": "A1",
  "a10": "A10",
  "a11": "A11",
  "a2": "A2",
  "a3": "A3",
  "a7": "A7",
  "a9": "A9",
  "c1": "C1",
  "db": "Custom rationale ({{db}})",
  "f1": "F1",
  "f2": "F2",
  "f3": "F3",
  "f7": "F7",
  "f9": "F9",
  "g1": "G1",
  "g10": "G10",
  "g11": "G11",
  "g12": "G12",
  "g13": "G13",
  "g14": "G14",
  "g2": "G2",
  "g3": "G3",
  "g4": "G4",
  "g6": "G6 ("copy-paste move" only)",
  "r2": "R2",
  "r3": "R3",
  "r4": "R4",
  "u5": "U5",
  "x3": "X3",
}
`;

exports[`modules/twinkleconfig Twinkle.config global variables Should match snapshot 4`] = `
[
  "db",
  "a1",
  "a2",
  "a3",
  "a7",
  "a9",
  "a10",
  "a11",
  "c1",
  "f1",
  "f2",
  "f3",
  "f7",
  "f9",
  "g1",
  "g2",
  "g3",
  "g4",
  "g6",
  "g10",
  "g11",
  "g12",
  "g13",
  "g14",
  "r2",
  "r3",
  "r4",
  "u5",
  "x3",
]
`;

exports[`modules/twinkleconfig Twinkle.config global variables Should match snapshot 5`] = `
{
  "a1": "A1",
  "a10": "A10",
  "a11": "A11",
  "a2": "A2",
  "a3": "A3",
  "a7": "A7",
  "a9": "A9",
  "c1": "C1",
  "c4": "C4",
  "db": "Custom rationale ({{db}})",
  "f1": "F1",
  "f11": "F11",
  "f2": "F2",
  "f3": "F3",
  "f4": "F4",
  "f5": "F5",
  "f6": "F6",
  "f7": "F7",
  "f8": "F8",
  "f9": "F9",
  "g1": "G1",
  "g10": "G10",
  "g11": "G11",
  "g12": "G12",
  "g13": "G13",
  "g14": "G14",
  "g2": "G2",
  "g3": "G3",
  "g4": "G4",
  "g5": "G5",
  "g6": "G6",
  "g7": "G7",
  "g8": "G8",
  "r2": "R2",
  "r3": "R3",
  "r4": "R4",
  "t5": "T5",
  "u1": "U1",
  "u2": "U2",
  "u5": "U5",
  "x3": "X3",
}
`;

exports[`modules/twinkleconfig Twinkle.config global variables Should match snapshot 6`] = `
[
  "db",
  "a1",
  "a2",
  "a3",
  "a7",
  "a9",
  "a10",
  "a11",
  "c1",
  "c4",
  "f1",
  "f2",
  "f3",
  "f4",
  "f5",
  "f6",
  "f7",
  "f8",
  "f9",
  "f11",
  "g1",
  "g2",
  "g3",
  "g4",
  "g5",
  "g6",
  "g7",
  "g8",
  "g10",
  "g11",
  "g12",
  "g13",
  "g14",
  "r2",
  "r3",
  "r4",
  "t5",
  "u1",
  "u2",
  "u5",
  "x3",
]
`;
